File: src/main/java/com/gpuflight/agent/publisher/PublisherFactory.java
package com.gpuflight.agent.publisher;

import com.gpuflight.agent.config.HttpConfig;
import com.gpuflight.agent.config.KafkaConfig;
import com.gpuflight.agent.config.PublisherConfig;

public class PublisherFactory {
    public static Publisher create(PublisherConfig config) {
        return switch (config) {
            case KafkaConfig kafka -> new KafkaPublisher(kafka.bootstrapServers());
            case HttpConfig http  -> new HttpPublisher(http);
        };
    }
}

File: src/main/kotlin/com/gpuflight/agent/publisher/PublisherFactory.kt
package com.gpuflight.com.gpuflight.agent.publisher

import com.gpuflight.com.gpuflight.agent.config.HttpConfig
import com.gpuflight.com.gpuflight.agent.config.KafkaConfig
import com.gpuflight.com.gpuflight.agent.config.PublisherConfig

object PublisherFactory {

    fun create(config: PublisherConfig): Publisher {
        return when (config) {
            is KafkaConfig -> KafkaPublisher(config.bootstrapServers)
            is HttpConfig -> HttpPublisher(config)
        }
    }
}